ui/src/pageLayout/components/NavMenuItem.tsx

```diff
@@ -1,48 +1,81 @@
 // Libraries
-import React, {SFC} from 'react'
+import React, {Component} from 'react'
 import {Link} from 'react-router'
 import classnames from 'classnames'
-import {get} from 'lodash'
 
 // Types
-import {IconFont} from 'src/clockface'
+import {IconFont, NavMenuType} from 'src/clockface'
 
-interface Props {
+interface PassedProps {
   icon: IconFont
   title: string
-  link: string
+  path: string
   children?: JSX.Element | JSX.Element[]
-  location: string
-  highlightPaths: string[]
+  active: boolean
+  className?: string
 }
 
-const NavMenuItem: SFC<Props> = ({
-  icon,
-  title,
-  link,
-  children,
-  location,
-  highlightPaths,
-}) => {
-  const parentPath = get(location.split('/'), '3', '')
-  const isActive = highlightPaths.some(path => path === parentPath)
-
-  return (
-    <div
-      className={classnames('nav--item', {active: isActive})}
-      data-testid={`nav-menu-item ${icon}`}
-    >
-      <Link className="nav--item-icon" to={link}>
-        <span className={`icon sidebar--icon ${icon}`} />
-      </Link>
-      <div className="nav--item-menu">
-        <Link className="nav--item-header" to={link}>
-          {title}
-        </Link>
-        {children}
+interface DefaultProps {
+  type?: NavMenuType
+  testID?: string
+}
+
+type Props = PassedProps & Partial<DefaultProps>
+
+class NavMenuItem extends Component<Props> {
+  public static defaultProps: DefaultProps = {
+    type: NavMenuType.RouterLink,
+    testID: 'nav-menu--item',
+  }
+
+  public render() {
+    const {
+      icon,
+      title,
+      path,
+      children,
+      active,
+      testID,
+      type,
+      className,
+    } = this.props
+
+    if (type === NavMenuType.RouterLink) {
+      return (
+        <div
+          className={classnames('nav--item', {
+            active,
+            [`${className}`]: className,
+          })}
+          data-testid={`${testID} ${title}`}
+        >
+          <Link className="nav--item-icon" to={path}>
+            <span className={`icon sidebar--icon ${icon}`} />
+          </Link>
+          <div className="nav--item-menu">
+            <Link className="nav--item-header" to={path}>
+              {title}
+            </Link>
+            {children}
+          </div>
+        </div>
+      )
+    }
+
+    return (
+      <div className={classnames('nav--item', {active})} data-testid={testID}>
```

Comment:
should the classname prop be able to be used in this case too?

Reply:
ah good catch